avocado/plugins/run.py

```diff
@@ -185,4 +186,13 @@ def run(self, args):
             log.error(e.message)
             sys.exit(exit_codes.AVOCADO_FAIL)
         job_instance = job.Job(args)
-        return job_instance.run()
+        job_run = job_instance.run()
+        result_dispatcher = ResultDispatcher()
+        if result_dispatcher.extensions:
+            # At this point job_instance doesn't have a single results attribute
+            # which is the end goal.  For now, we pick any of the plugin classes
+            # added to the result proxy.
+            if len(job_instance.result_proxy.output_plugins) > 0:
+                result = job_instance.result_proxy.output_plugins[0]
+                result_dispatcher.map_method('render', result, job_instance)
```

Comment:
I'm not sure how do you want to continue on this, but I'd suggest making the base `Result` class part of the `result_proxy` and each called method should receive it as an argument. Then it would be explicit what is given in `Result` classes and what are the custom methods/variables of the custom result plugin. Additionally only one copy of `Result` would be stored in the memory.

Doing this does not require much changes, so I'm wondering if we shouldn't do this as part of this PR, but maybe you thought of this for the stage2 (or you have different approach).